### ISSUE TITLE
Unblock dev by using cifar classifier as toy model

### ISSUE THREAD

**Issue by caseyfitz** [drivendataorg/zamba#1]
## The Problem
Now that we have simple functionality, we need to hone in on specific features. To do this efficiently, we need
* a lightweight model pipeline
* a pipeline similar enough to the primatrix classifier so that core functionality can be developed
* a pipeline simple enough so that development can proceed at a reasonable pace

## A Solution
Building a convolutional network to classify the cifar10 data fits the above criteria. The cifar 10 problem is to classify 32x32 pixel RBG images into one of 10 classes. It is a dataset often used in academics, but I think can be very useful for application development as well, at least in our case.

The idea is to use cifar10 as the context, and a simple convolutional net as the "default model" (eventually dmytro's) that we can use to develop the `predict`, `train`, and `tune` functionality. Since the inputs (images) and outputs (row of probs for one image) are similar, minimizing the code's bias towards cifar should be less complicated than if we keep using random number data and non-predictors as a development model.

The **goal** is to frame as many pri-martrix problems as we can as cifar problems. My *claim* is that there are a non-trivial amount of pri-matrix problems that allow this framing. For example,
1. Handling real **input** directories, pre-processing their contents into temporary storage, and sending them into a model for prediction.
2. Handling **output** that is similarly structured to the chimps output, i.e., csv files indexed by a filename with class probabilities as columns.
3. Training, tuning, saving, and loading **functionality for classifiers**, not just tensorflow graphs (although that was a good start).
4. Development of any** additional performance metrics**, such as loss over time, validation scoring, etc.

As the [tensorflow advanced tutorial](https://www.tensorflow.org/tutorials/deep_cnn)  on cifar says,
> The reason CIFAR-10 was selected was that it is complex enough to exercise much of TensorFlow's ability to scale to large models. At the same time, the model is small enough to train fast, which is ideal for trying out new ideas and experimenting with new techniques.

In short, this approach seems to me a good solution to asynchronously working on the codebase with dmytro before his models are ready for integration.

## Advantages
* The above-referenced tutorial comes with a lot of code that can be edited and used to speed up development of our project, such as network architecture, training criteria, data-fetching scripts, pre-processing tasks, and various other "advanced" features that we can consciously decide to develop or not as we work on the tool.
* The fact that inputs to the dmytro model are images and these data are images will allow us to develop a lot of the preprocessing pipeline (except the parts that convert the videos into images).
* We can experiment with some of the ensembling that we'll eventually need to handle, such as using tensorflow and xgboost to generate predictions.

## Disadvantages
* Will inevitably have to deal with some idiosyncrasies of the cifar10 data, but given its simplicity and the fact that we have lots of starter code for manipulating it, this shouldn't set us back too much.

## Timeline
Since the code skeleton is now fairly operational, I think it's reasonable to shoot for integrating this by the end of the week. End of first rev at the latest.
